File: 0058-length-of-last-word/0058-length-of-last-word.cpp
class Solution {
public:
    int lengthOfLastWord(string s) {
        int countSpaces=0;
        for(int i=s.length()-1;i>=0;i--){
            if(s[i]==' '){
                countSpaces++;
            }
            else{
                break;
            }
        }
        int count=0;
        for(int i=s.length()-1-countSpaces;i>=0;i--){
            if(s[i]!=' '){
                count++;
            }
            if(s[i]==' ' && s[i+1]!=' '){
                break;
            }
        }
        return count;
    }
};

File: 0058-length-of-last-word/0058-length-of-last-word.rs
impl Solution {
    pub fn length_of_last_word(s: String) -> i32 {
        
        let trimmed = s.trim();
        let words: Vec<&str> = trimmed.split_whitespace().collect();

        if let Some(last_word) = words.last() {
            return last_word.len() as i32;
        } else {
            return 0;
        }        
    
    }
}